docs/src/main/docs/tutorials/testing-rpc-service.md

```diff
@@ -0,0 +1,204 @@
+---
+layout: docs
+title: Testing an RPC service
+section: tutorials
+permalink: tutorials/testing-rpc-service
+---
+
+# Tutorial: Testing an RPC service
+
+This tutorial will show you how to write a unit test for an RPC service using an
+in-memory channel and client.
+
+This tutorial is aimed at developers who:
+
+* are new to Mu-Scala
+* have some understanding of Protobuf and `.proto` file syntax
+* have read the [Getting Started guide](../getting-started)
+* have followed the [gRPC server and client tutorial](grpc-server-client)
+
+## Service definition
+
+Let's use the following service definition. (For an explanation of how to create
+service definition, check out the [RPC service definition with Protobuf
+tutorial](service-definition/protobuf).)
+
+A client sends a `HelloRequest` containing a name, and the server responds with
+a greeting and an indication of whether it is feeling happy or not.
+
+```scala mdoc:silent
+import higherkindness.mu.rpc.protocol._
+
+object hello {
+  case class HelloRequest(@pbdirect.pbIndex(1) name: String)
+  case class HelloResponse(@pbdirect.pbIndex(1) greeting: String, @pbdirect.pbIndex(2) happy: Boolean)
+
+  @service(Protobuf, Identity, namespace = Some("com.example"), methodNameStyle = Capitalize)
+  trait Greeter[F[_]] {
+    def SayHello(req: HelloRequest): F[HelloResponse]
+  }
+
+}
+```
+
+## Service implementation
+
+Here's the implementation we want to test.
+
+```scala mdoc:silent
+import cats.Applicative
+import cats.syntax.applicative._
+import hello._
+
+class HappyGreeter[F[_]: Applicative] extends Greeter[F] {
+
+  def SayHello(req: HelloRequest): F[HelloResponse] =
+    HelloResponse(s"Hello, ${req.name}!", happy = true).pure[F]
+
+}
+```
+
+We're going to write a test to check that the service is always happy.
+
+## cats-effect implicits
+
+In our test we'll use cats-effect `IO` as our concrete effect monad.
+
+We need to provide a couple of implicits to make that work:
+
+```scala mdoc:silent
+import cats.effect.{IO, ContextShift, Timer}
+import scala.concurrent.ExecutionContext.Implicits
+
+trait CatsEffectImplicits {
+
+  val EC: scala.concurrent.ExecutionContext = Implicits.global
```

Comment:
:art:, since you're not making use of the ec implicitly, you can do:

```suggestion
import scala.concurrent.ExecutionContext

trait CatsEffectImplicits {

  val EC: ExecutionContext = ExecutionContext.global
```